### PR TITLE
Add Kelvinlet deformations

### PR DESCRIPTION
**Pull request by SooryaN**
Implements the Kelvinlets sculpting brushes

[Demo here](https://streamable.com/77f8xf)

#### Check all that apply (change to `[x]`)
- [x] All changes meet [libigl style-guidelines](https://libigl.github.io/style-guidelines/).
- [x] Adds new .cpp file.
- [ ] Adds corresponding unit test.
- [ ] This is a minor change.

### CODE REVIEW

File: include/igl/kelvinlets.cpp [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+// This Source Code Form is subject to the terms of the Mozilla Public License
+// v. 2.0. If a copy of the MPL was not distributed with this file, You can
+
+#include "kelvinlets.h"
+#include "PI.h"
+#include "parallel_for.h"
+
+constexpr double POISSON_RATIO = 0.5;
+constexpr double SHEAR_MODULUS = 1;
+constexpr double a = 1 / (4 * igl::PI * SHEAR_MODULUS);
```
Comment (jdumas) [inline include/igl/kelvinlets.cpp:14]: Can you define those locally, so as not to pollute the global namespace?

File: include/igl/kelvinlets.cpp [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+constexpr double b = a / (4 * (1 - POISSON_RATIO));
+constexpr double c = 2 / (3 * a - 2 * b);
+
+template<typename Derivedx,
+         typename Derivedx0,
+         typename Derivedf,
+         typename DerivedF,
+         typename Scalar>
+IGL_INLINE auto kelvinlet_evaluator(const Scalar dt,
```
Comment (jdumas) [inline include/igl/kelvinlets.cpp:23]: Enclose within the `igl` namespace + maybe add some doc?

File: include/igl/kelvinlets.cpp [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+          kp.w[2] * kelvinlet(kp.ep[2])) *
+         20;
+};
+
+template<typename Scalar,
+         typename Derivedx,
+         typename Derivedx0,
+         typename Derivedf,
+         typename DerivedF>
+IGL_INLINE void integrate(const Scalar t,
```
Comment (jdumas) [inline include/igl/kelvinlets.cpp:105]: Can you add some doc to this function?

**Review by jdumas — COMMENTED** [review]
Nice!

Just got minor comments regarding the .cpp file, since libigl can be used in header-only mode, we need to protect the content of the .cpp file from polluting other headers.

File: include/igl/kelvinlets.cpp (hunk 1 above)
Comment (SooryaN) [reply include/igl/kelvinlets.cpp]: Moved them

File: include/igl/kelvinlets.cpp (hunk 2 above)
Comment (SooryaN) [reply include/igl/kelvinlets.cpp:31]: Done

File: include/igl/kelvinlets.cpp (hunk 3 above)
Comment (SooryaN) [reply include/igl/kelvinlets.cpp:143]: Done. Let me know if that's good enough

**Review by jdumas — APPROVED** [review]

**Comment by jdumas** [comment]
@SooryaN would you be able to add a tutorial entry to the website (https://github.com/libigl/libigl.github.io), now that the feature has been integrated?

**Comment by SooryaN** [comment]
@jdumas, sure, I'll add it in this week

**Pull request merged**